### ISSUE TITLE
[iOS11]: HTTPS Connection: Keep-Alive does not work properly

### ISSUE THREAD

**Issue by MorpheusDe97** [silkimen/cordova-plugin-advanced-http#115]
With iOS 11, we got a problem using the cordova-plugin-advanced-http. 

We want the App to produce a persistant connection to a server with just one certificate handshake. Thats why we set the keep-alive-header for each request. 

      let headers = {
        Connection: "keep-alive",
      };

But what we get is, if the App sends several requests with different nodes, like
https://{SERVER_IP}/Node1
https://{SERVER_IP}/Node2
https://{SERVER_IP}/Node3

What we now get is: The first request is kept alive, until the server closes the connection because of an arriving keep-alive-timeout. 

The app waits, until the first connection was closed and starts the next request, allthough we thought that the client sends the other requests immediately after the first response. 

For android, it works pretty fine, but in iOS, the keep-alive doesn't work as expected. 

Is there any other flag or function that we have to set or to call, so that the keep-alive-connection works in iOS?

**Comment by MorpheusDe97** [comment]
@silkimen Do you understand my problem? Can you reproduce it using iOS? 

If there is anything unclear for you, please feel free to ask me!

**Comment by karisdft4567** [comment]
thats the problem with cross platform

**Comment by silkimen** [comment]
I understand your problem, but I'm very busy, atm. I didn't reproduce it, yet. Maybe someone can fix this and send me a PR for this one?

**Comment by silkimen** [comment]
Please try version 2.0.6+. Closing this, feel free to open a new issue, if this won't solve your problem!

**Comment by chrisjdev** [comment]
Looks like this should be reopened for the latest code. At some point, the fix was reverted. I tried reapplying the same fix to the latest code, but it looks like a memory leak fix in pull request #399 ensures that the socket is closed after every request. Removing that change fixes this issue and works for me, but I'm betting it would re-introduce issue #197. 

Looking at SM_AFHTTPSessionManager.h, it even has a warning:
> Managers for background sessions must be owned for the duration of their use. This can be accomplished by creating an application-wide or shared singleton instance.

So I think the code was on the right track keeping a reference to the manager with the plugin and creating it once in pluginInitialize. Looks like the plugin code is all running on the main thread, so I'm not sure where there might be a race condition. I'll try to get the e2e tests running on my machine to see what fails.
